src/cluster_legacy.c

```diff
@@ -3203,10 +3178,21 @@ int clusterProcessPacket(clusterLink *link) {
             }
         }
 
+        /* Explicitly check for a replication loop before attempting the replication
+         * chain folding logic. */
+        if (myself->replicaof && myself->replicaof->replicaof && myself->replicaof->replicaof != myself) {
```

Comment:
Ah, this is becoming so complex.


Where do we handle the cyclic scenario?
```
myself->replicaof && myself->replicaof->replicaof && myself->replicaof->replicaof == myself
```

Reply:
This is not for the cyclic replication handling. This is the replication chain reduction/compression logic. 

We handle/prevent the cyclic by allowing primaries only to enter `clusterUpdateSlotsConfigWith`

Reply:
BTW, this is not new logic. It is shifted downwards.